File: batchbhavna/FirstApplication/app/src/main/java/com/example/firstapplication/fragments/FirstFragment.java
package com.example.firstapplication.fragments;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

import com.example.firstapplication.R;


public class FirstFragment extends Fragment {
Button movetobtn,checkbtn;

RadioGroup radioGroup;

    private static final String TAG = "FirstFragment";
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        Log.e(TAG, "onCreateView: " );
        return inflater.inflate(R.layout.fragment_first, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Log.e(TAG, "onViewCreated: " );
        movetobtn = view.findViewById(R.id.movetofirst);
       radioGroup = view.findViewById(R.id.radiogroup);
       checkbtn = view.findViewById(R.id.check);

       checkbtn.setOnClickListener(new View.OnClickListener() {
           @Override
           public void onClick(View v) {
               int selecteid = -1;
                selecteid = radioGroup.getCheckedRadioButtonId();

              RadioButton radioButton = view.findViewById(selecteid);



               if(selecteid < 0)
               {
                   Toast.makeText(getContext(),"select atleast one option",Toast.LENGTH_SHORT).show();
               }else
               {
                   String text = radioButton.getText().toString();
                   if(text.equals("less18")){
                       Toast.makeText(getContext(), "You are not eligible for voting", Toast.LENGTH_SHORT).show();
                   }else
                   {
                       Toast.makeText(getContext(), "You are  eligible for voting", Toast.LENGTH_SHORT).show();

                   }
               }
           }
       });
        movetobtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
              Fragment fragment = new SecondFragment();
              Bundle args = new Bundle();
              args.putString("first","Data from Firstfragment");
              fragment.setArguments(args);


                FragmentManager fragmentManager = getFragmentManager();
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                fragmentTransaction.replace(R.id.fragmentcontainer,fragment);
                fragmentTransaction.commit();
            }
        });
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.e(TAG, "onDestroy: " );
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Log.e(TAG, "onAttach: " );
    }

    @Override
    public void onStart() {
        super.onStart();
        Log.e(TAG, "onStart: " );
    }

    @Override
    public void onResume() {
        super.onResume();
        Log.e(TAG, "onResume: " );
    }

    @Override
    public void onPause() {
        super.onPause();
        Log.e(TAG, "onPause: " );
    }

    @Override
    public void onStop() {
        super.onStop();
        Log.e(TAG, "onStop: " );
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.e(TAG, "onDestroyView: " );
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.e(TAG, "onDetach: " );
    }
}

File: 4pmclass/MyFirstApplication/app/src/main/java/com/example/myfirstapplication/FirstFragment.kt
package com.example.myfirstapplication

import android.content.Context
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button


class FirstFragment : Fragment() {

lateinit var button: Button

var TAG:String = "First Fragment"
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        Log.e(TAG,"OnCreadtVeiw Called")
        return inflater.inflate(R.layout.fragment_first, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.e(TAG,"OnCreadtVeiwCreadted Called")
        button = view.findViewById(R.id.move_to_seocndFragment);
        button.setOnClickListener {
            val fragment = SecondFragment()
            var bundle = Bundle()
            bundle.putString("first","Data from First Fragment")
            fragment.arguments = bundle
            val fragmentManager = fragmentManager
            val fragmentTraction = fragmentManager!!.beginTransaction();
            fragmentTraction.replace(R.id.fragmentcontainer,fragment);
            fragmentTraction.commit()
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        Log.e(TAG,"OnAttach Called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e(TAG,"OnDestroy");
    }

    override fun onStart() {
        super.onStart()
        Log.e(TAG,"OnStart");
    }

    override fun onResume() {
        super.onResume()
        Log.e(TAG,"OnDResumen");
    }

    override fun onPause() {
        super.onPause()
        Log.e(TAG,"OnPause");
    }

    override fun onStop() {
        super.onStop()
        Log.e(TAG,"OnStopd");
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.e(TAG,"OnDestroyView");
    }

    override fun onDetach() {
        super.onDetach()
        Log.e(TAG,"OnDettach");
    }
}